doc/manipulating_visualizing/data_preparation.rst

```diff
@@ -310,51 +350,41 @@ Extraction of signals from regions:\  :class:`NiftiLabelsMasker`, :class:`NiftiM
 
 The purpose of :class:`NiftiLabelsMasker` and :class:`NiftiMapsMasker` is to
 compute signals from regions containing many voxels. They make it easy to get
-these signals once you have an atlas or a parcellation.
+these signals once you have an atlas or a parcellation into brain regions.
 
 Regions definition
 ------------------
 
 Nilearn understands two different ways of defining regions, which are called
-labels and maps, handled respectively by :class:`NiftiLabelsMasker` and
-:class:`NiftiMapsMasker`.
+labels and maps, handled by :class:`NiftiLabelsMasker` and
+:class:`NiftiMapsMasker`, respectively.
 
 - labels: a single region is defined as the set of all the voxels that have a
-  common label (usually an integer) in the region definition array. The set of
-  regions is defined by a single 3D array, containing at each location the label
-  of the region the voxel is in. This technique has one big advantage: the
-  amount of memory required is independent of the number of regions, allowing
-  for representing a large number of regions. On the other hand, there are
-  several contraints: regions cannot overlap, and only their support could be
-  represented (no weighting).
+  common label (e.g., anatomical brain region definitions as integers)
+  in the region definition array. The set of
+  regions is defined by a single 3D array, containing a voxel-wise
+  dictionary of label numbers that denote what
+  region a given voxel belongs to. This technique has a big advantage: the
+  required memory load is independent of the number of regions, allowing
+  for a large number of regions. On the other hand, there are
+  several disadvantages: regions cannot spatially overlap
+  (i.e., regions cannot overlap)
```

Comment:
"regions cannot spatially overlap" is quite redundant with "(i.e., regions cannot overlap)"